page-objects/pages/TASKListPage.js

```diff
@@ -160,67 +147,74 @@ async function clickDeleteTaskButtonInPopup() {
 
 async function clickAtTask(taskName) {
 	await waitHelpers.waitUntilNuxtClientLoads();
-	await driver.pause(5000);
-	let clickOnThatTask = (await getTaskFromNuxtClient(taskName)).toString();
-	await elementHelpers.scrollToElement(clickOnThatTask);
-	await elementHelpers.clickAndWait(clickOnThatTask);
+	let clickOnThatTask = (await getTaskFromTaskOverview(taskName));
+	if (clickOnThatTask === undefined){
+		let isTaskClickable = new Boolean (false);
+		while (!isTaskClickable){
+			elementHelpers.scrollToElement(clickOnThatTask);
+			await waitHelpers.waitUntilElementIsVisible(clickOnThatTask);
+			if (clickOnThatTask.isDisplayedInViewport()){
+				isTaskClickable = true;
+			}
+		}
+	}
+	if (clickOnThatTask.isDisplayedInViewport()) {
+		await elementHelpers.clickAndWait(clickOnThatTask);
```

Comment:
Maybe I'm wrong, but can't we add 'await elementHelpers.clickAndWait(clickOnThatTask);' below 'if (clickOnThatTask.isDisplayedInViewport()){
				isTaskClickable = true;'?